### PR TITLE
Get data from device for email settings page

### PR DESCRIPTION
**Pull request by alistairjcbrown**
**Reviewer:** @GioSensation 
**Asana:** https://app.asana.com/0/0/1201023447886986/f

## Description
Wire up getting basic user data from device to show on web app settings page. Splits out `setupSettingsPage` specific functionality from existing `setupAutofill`

## Steps to test
See task linked above

### CODE REVIEW

**Comment by GioSensation** [comment]
@alistairjcbrown My understanding is that currently this is only supposed to work on extensions, right? Should I hold off reviewing?

**Comment by alistairjcbrown** [comment]
@GioSensation I believe the aim is for the web app settings page to be usable on any device with email autofill enabled, which would cover Apple devices and Android devices too. There currently aren't PRs to update those device types (just one for extensions) -- does that block reviewing this?
And also, could this land without those devices being updated to provide support (and be updated in the future)?

**Comment by GioSensation** [comment]
It doesn't stop reviewing per se, but it stops releasing. We would either merge to a separate release branch or provide failsafes so that other platforms don't error (like returning early instead of making unexpected calls). At least on Android I think that sending unexpected data might stop autofill from working altogether. On Apple devices it just logs errors, IIRC.

I think the plan was to release this stuff asap everywhere, wasn't it?

File: src/Form/Form.js [hunk 1]
```diff
@@ -120,15 +120,15 @@ class Form {
...
+        this.execOnInputs((input) => this.removeInputHighlight(input), dataType)
     }
 
     removeInputDecoration (input) {
         removeInlineStyles(input, getIconStylesBase(input, this))
         input.removeAttribute(ATTR_AUTOFILL)
     }
     removeAllDecorations () {
-        this.execOnInputs(this.removeInputDecoration)
+        this.execOnInputs((input) => this.removeInputDecoration(input))
```
Comment (alistairjcbrown) [inline src/Form/Form.js:131]: Need to make sure these functions are called with the right context bound as they use `this` internally

**Review by GioSensation — APPROVED** [review]
Check that possible conflict mishap, but otherwise this look good to me. I've checked it in the extension and that on macOS it doesn't break the flow. :shipit:

File: src/Form/inputTypeConfig.js [hunk 2]
```diff
@@ -11,7 +11,7 @@ const getDaxImg = isDDGApp || isFirefox ? daxBase64 : chrome.runtime.getURL('img
 /**
  * Get the icon for the identities (currently only Dax for emails)
  * @param {HTMLInputElement} input
- * @param device
+ * @param {import("./Form").Form} form
```
[the remark this answers is not in the document]
Comment (alistairjcbrown) [reply src/Form/inputTypeConfig.js:14]: @GioSensation The second parameter for this function a `Form` type object  that we destructure `device` from (line 17 below) -- the bugfix I made in `src/Form/Form.js` was to fix where we were passing `undefined` for this parameter.

**Pull request merged**